p8015\_hw1\_tl2882
================
Tian Li
2018-09-14

Problem 1
=========

Create a data frame
-------------------

``` r
set.seed(1)

df = tibble(
  random_sample = runif(10, min = 0, max = 5),
  vec_logical = (random_sample > 2),
  vec_char = c("A", "B", "C", "D", "E", "a", "b", "c", "d", "e"),
  vec_factor = factor(c("A", "B", "C", "D", "E", "a", "b", "c", "d", "e"))
)
```

Try to take the mean
--------------------

``` r
mean(df$random_sample)
```

    ## [1] 2.757569

``` r
mean(df$vec_logical)
```

    ## [1] 0.6

``` r
mean(df$vec_char)
```

    ## Warning in mean.default(df$vec_char): argument is not numeric or logical:
    ## returning NA

    ## [1] NA

``` r
mean(df$vec_factor)
```

    ## Warning in mean.default(df$vec_factor): argument is not numeric or logical:
    ## returning NA

    ## [1] NA

1.Succeed: normal arithmetical mean
2.Succeed: "ture" means "1", and "false" means "0"
3.Fail: only numeric or logical vector can calculate the mean
4.Fail: only numeric or logical vector can calculate the mean

Convert variables
-----------------

### Applies the as.numeric function

``` r
as.numeric(df$vec_logical)
```

    ##  [1] 0 0 1 1 0 1 1 1 1 0

``` r
as.numeric(df$vec_char)
```

    ## Warning: NAs introduced by coercion

    ##  [1] NA NA NA NA NA NA NA NA NA NA

``` r
as.numeric(df$vec_factor)
```

    ##  [1]  2  4  6  8 10  1  3  5  7  9

What happens? two warnings.

1.  Succeed ("ture" is converted to "1", and "false" is converted to "0")
2.  Fail: can't be converted to numeric
3.  Succeed : can be converted to numbers which represent their alphabetical order
